nix-darwin/default.nix

```diff
@@ -14,6 +14,23 @@ in
   imports = [ ../nixos/common.nix ];
 
   config = lib.mkMerge [
+    {
+      home-manager.sharedModules = [
+        ({
+
+          targets.darwin.linkApps.enable = lib.mkDefault false;
```

Comment:
Should take after https://github.com/nix-community/home-manager/pull/8031 to use `stateVersion` to default to the previous behavior for existing configs and future versions enable the new behavior to leverage `nix-darwin` application installation.